### PR TITLE
fix(background): handle unpeered wallets; retry without waiting interval

### LINKED ISSUE #394: [BUG] No warning when sending to real wallet from rafiki.money
### Steps to reproduce

1. Go to https://roamancing.com/naturallyours/ (uses fynbos wallet)
2. Setup wallet with rafiki.money
3. No payments can go through

### Expected result

Payments don't go through but there should be an indication in extension, and extension should stop trying to pay.

### Actual result

Extension keeps trying payments indiscrimately.

### Screenshots or videos

image

image

### Additional context

_No response_

### Operating system

Linux

### Operating system version

_No response_

### Browsers

Chrome, Firefox

### Browser version

_No response_

### Extension version

alpha2

### PR DESCRIPTION
**Pull request by dianafulga**
- Closes #394 
- Closes https://github.com/interledger/web-monetization-extension/issues/469
- Closes https://github.com/interledger/web-monetization-extension/issues/296
- Part of https://github.com/interledger/web-monetization-extension/issues/147

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">6f618ad81732606402fc13f2dadffc5499116b2d</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/10183296402">Run #10183296402</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/26646789178/artifacts/1760824105">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/26646789178/artifacts/1760824108">Download</a></td></tr>
</tbody>
</table>

File: src/_locales/en/messages.json [hunk 1]
```diff
@@ -40,6 +40,12 @@
   "pay_error_notEnoughFunds": {
     "message": "Not enough funds to facilitate payment."
   },
+  "pay_error_invalidReceivers": {
+    "message": "[TODO] We cannot send money (probable cause: un-peered wallets)"
+  },
+  "pay_error_notMonetized": {
+    "message": "[TODO] This website is not monetized."
+  },
```
Comment (raducristianpopa) [inline src/_locales/en/messages.json:48]: @RabebOthmani

File: src/_locales/en/messages.json [hunk 2]
```diff
@@ -85,5 +91,8 @@
   },
   "connectWallet_error_invalidClient": {
     "message": "Failed to connect. Please make sure you have added the public key to the correct wallet address."
+  },
+  "home_warn_invalidLinks": {
+    "message": "[TODO] Has invalid links. (UI)"
```
Comment (raducristianpopa) [inline src/_locales/en/messages.json:96]: @RabebOthmani

Comment (raducristianpopa) [reply src/_locales/en/messages.json:96]: image

File: src/_locales/en/messages.json (hunk 1 above)
Comment (raducristianpopa) [reply src/_locales/en/messages.json:48]: image

**Comment by RabebOthmani** [comment]
@raducristianpopa are these still waiting for me?

**Comment by raducristianpopa** [comment]
> @raducristianpopa are these still waiting for me?

Yes. We added them here as well so you can see where and how they are displayed.

**Comment by RabebOthmani** [comment]
@raducristianpopa few thoughts before getting into the copy: 
Across the extension, we use payment. Do we want to keep with payments (Unable to process a payment) or use sent money (unable to send money) ? 
Thinking about the average user: will they understand what a peered or unpeered wallets mean? I think it's better to say we can't process the payment, you might need to check with your provider for more details

**Comment by RabebOthmani** [comment]
For the error message when a website is not monetized, let's go with "This site is not web monetized" (using web monetized on purpose to keep it in the context of web monetization)

**Comment by raducristianpopa** [comment]
>Across the extension, we use payment. Do we want to keep with payments (Unable to process a payment) or use sent money (unable to send money) ?
Like you mentioned, we are using payment across the extension. Let's stick with this.

>Thinking about the average user: will they understand what a peered or unpeered wallets mean? I think it's better to say we can't process the payment, you might need to check with your provider for more details

No, we should not mention peering for the end user. What I am not sure about is this part: `you might need to check with your provider for more details`. The user doesn't have any details about who the receiver is since we are not showing them the wallet addresses that we found in the page. What is the user going to tell to the support them?

**Comment by RabebOthmani** [comment]
> No, we should not mention peering for the end user. What I am not sure about is this part: you might need to check with your provider for more details. The user doesn't have any details about who the receiver is since we are not showing them the wallet addresses that we found in the page. What is the user going to tell to the support team for example?

What I'm trying to communicate is the payment failed but that's not the extension's fault. How about : We are unable to process the payment due to issues beyond our control.

**Comment by RabebOthmani** [comment]
The last one, what do we mean by has invalid linked? Is that when it's a partial problem?

**Comment by raducristianpopa** [comment]
> The last one, what do we mean by has invalid linked? Is that when it's a partial problem?

The `has invalid links` placeholder that we have, will be displayed to the user when the extension can not send money to the wallet addresses that we're found in the page (in our case, mostly because on the peering). It's not a partial problem. For the partial problem - when we can send to at least one wallet address from the page, we are not going to show the warning.

**Comment by RabebOthmani** [comment]
We discussed all these scenarios from our perspective. Now that we're working on the message, I'm wondering if from the user's perspective, they are distinct scenarios. The one thing I could think about that would make this last case different (invalid links) is that potentially those links may change in the future. We could say " Currently, we are unable to process payments for this website. This might change in the future should you wish to keep the continuous payment going." What do you think @raducristianpopa

**Comment by RabebOthmani** [comment]
Final decision per the call: At the moment, you cannot pay this website.

File: src/_locales/en/messages.json [hunk 3]
```diff
@@ -40,6 +40,12 @@
   "pay_error_notEnoughFunds": {
     "message": "Not enough funds to facilitate payment."
   },
+  "pay_error_invalidReceivers": {
+    "message": "[TODO] We cannot send money (probable cause: un-peered wallets)"
```
Comment (sidvishnoi) [inline src/_locales/en/messages.json:44]:
```suggestion
    "message": "At the moment, you cannot pay this website.",
    "description": "We cannot send money (probable cause: un-peered wallets)"
```

File: src/_locales/en/messages.json [hunk 4]
```diff
@@ -40,6 +40,12 @@
...
+  },
+  "pay_error_notMonetized": {
+    "message": "[TODO] This website is not monetized."
```
Comment (sidvishnoi) [inline src/_locales/en/messages.json:47]:
```suggestion
    "message": "This website is not monetized."
```

File: src/popup/pages/Home.tsx [hunk 5]
```diff
@@ -68,8 +70,16 @@ export const Component = () => {
...
+  if (hasAllSessionsInvalid) return <AllSessionsInvalid />
+
   return (
     <div className="space-y-8">
+      {/*
+        // TO DO: Check if still needed after discussing
+        // https://github.com/interledger/web-monetization-extension/issues/471
+        {hasAllSessionsInvalid && (
+        <WarningMessage warning={t('home_warn_invalidLinks')} />
+      )} */}
```
Comment (sidvishnoi) [inline src/popup/pages/Home.tsx:82]: Let's remove this for now. Will see when we change UI.
```suggestion
```

File: src/popup/pages/Home.tsx [hunk 6]
```diff
@@ -68,8 +70,16 @@ export const Component = () => {
     return <SiteNotMonetized />
   }
 
+  if (hasAllSessionsInvalid) return <AllSessionsInvalid />
```
Comment (sidvishnoi) [inline src/popup/pages/Home.tsx:73]: Consistency with other conditions.

```suggestion
  if (hasAllSessionsInvalid) {
    return <AllSessionsInvalid />
  }
```

**Review by sidvishnoi — CHANGES_REQUESTED** [review]
The design looks a bit off at the moment.



(And some tiny code changes suggested)

File: src/_locales/en/messages.json [hunk 7]
```diff
@@ -85,5 +91,8 @@
   },
   "connectWallet_error_invalidClient": {
     "message": "Failed to connect. Please make sure you have added the public key to the correct wallet address."
+  },
+  "allInvalidLinks_state_text": {
+    "message": "At the moment, you cannot pay this website."
```
Comment (raducristianpopa) [inline src/_locales/en/messages.json:96]:
```suggestion
    "message": "At the moment, you can not pay this website."
```

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**